src/common.c

```diff
@@ -444,7 +444,7 @@ char *get_temp_filename(const char *prefix)
 	}
 	lp = strlen(prefix);
 	result = malloc(lt + lp + 8);
-	strncpy(result, tmpdir, lt);
+	strncpy(result, tmpdir, sizeof(result) -1);
```

Comment:
It is not correct to use `sizeof` on an allocated pointer. On a 64 bit platform, `sizeof(result)` will be 8 and on 32 bit 4 but not the size of the allocated buffer. Even if it would return the correct size for `result` - which it doesn't - it might copy more than what should be copied. I know about these compiler warnings about sizes depending on the source buffer but we need to find a different way to silence it. The target buffer is DEFINITELY larger than the source buffer, unless the allocation fails.